flow/project.py

```diff
@@ -2932,12 +2996,24 @@ class MyProject(FlowProject):
             choices=['none', 'by-job', 'cyclic', 'random'],
             default=None,
             help="Specify the execution order of operations for each execution pass.")
+        execution_group.add_argument(
+            '--ignore-conditions',
+            type=str,
+            choices=['none', 'pre', 'post', 'all'],
+            default='none',
+            help="Specify conditions to ignore for eligibility check.")
         parser_run.set_defaults(func=self._main_run)
 
         parser_script = subparsers.add_parser(
             'script',
             parents=[base_parser],
         )
+        parser_script.add_argument(
+            '--ignore-conditions',
+            type=str,
+            choices=['none', 'pre', 'post', 'all'],
+            default='none',
+            help="Specify conditions to ignore for eligibility check.")
```

Comment:
Since the Python code is being changed to use the enum instead of strings, you'll need to switch these parsers to translate the strings to the enums (ideally using a custom [`argparse.Action`](https://docs.python.org/3/library/argparse.html#argparse.Action)).